app/src/custom_hpa/core.clj

```diff
@@ -0,0 +1,22 @@
+(ns custom-hpa.core
+  (:require [custom-hpa.helpers.env :refer [int-env]]
+            [custom-hpa.clients.kube :as kube]
+            [custom-hpa.web.server :as web-server]
+            [custom-hpa.control-loop :as control-loop]
+            [taoensso.timbre :as logger]
+            [custom-hpa.metric.core :as metric]
+            [clojure.string :as str])
+  (:gen-class))
+
+(defn -main
+  "Initializes core components and starts the custom HPA"
+  [& args]
+  (let [deployment (System/getenv "DEPLOYMENT")
+        deployment-namespace (System/getenv "NAMESPACE")
+        dry-run (= "true" (str/trim (System/getenv "DRY_RUN")))]
+    (logger/info "Custom HPA started")
+    (logger/set-level! (keyword (or (System/getenv "LOG_LEVEL") "info")))
+    (logger/swap-config! assoc-in [:timestamp-opts :pattern] "yyyy-MM-dd'T'HH:mm:ss.SSSZ")
+    (kube/init deployment deployment-namespace dry-run)
+    (web-server/start (or (int-env "PORT") 3000))
+    (control-loop/start (metric/default-provider))))
```

Comment:
I agree, I didn't think about using the code outside of the project.
I've refactored the code so that the control loop now receive the deployment and the namespace 
see https://github.com/nanit/kubernetes-custom-hpa/pull/1/commits/75293296bcc361e4df1394ee59506dd9b99600c0